programs/drift/src/state/fulfillment_params/openbook_v2.rs

```diff
@@ -0,0 +1,506 @@
+    pub openbook_v2_base_vault: &'a AccountInfo<'b>,
+    pub openbook_v2_quote_vault: &'a AccountInfo<'b>,
+    pub base_market_vault: Box<Account<'b, TokenAccount>>,
+    pub quote_market_vault: Box<Account<'b, TokenAccount>>,
+    pub token_program: Program<'b, Token>,
+    pub system_program: Program<'b, System>,
+    pub signer_nonce: u8,
+    pub now: i64,
+}
+
+// TODO
+impl<'a, 'b> OpenbookV2FulfillmentParams<'a, 'b> {
+    #[allow(clippy::type_complexity)]
+    pub fn new<'c: 'b>(
+        account_info_iter: &'a mut std::iter::Peekable<std::slice::Iter<'c, AccountInfo<'b>>>,
+        state: &State,
+        base_market: &SpotMarket,
+        quote_market: &SpotMarket,
+        now: i64,
+    ) -> DriftResult<Self> {
+        let account_info_vec = account_info_iter.collect::<Vec<_>>();
+        let account_infos = array_ref![account_info_vec, 0, 14];
+        let [
+            openbook_v2_fulfillment_config,
+            drift_signer,
+            openbook_v2_program,
+            openbook_v2_market,
+            openbook_v2_market_authority,
+            openbook_v2_event_heap,
+            openbook_v2_bids,
+            openbook_v2_asks,
+            openbook_v2_base_vault,
+            openbook_v2_quote_vault,
+            base_market_vault,
+            quote_market_vault,
+            token_program,
+            system_program
+            ] = account_infos;
+        let openbook_v2_fulfillment_config_loader: AccountLoader<OpenbookV2FulfillmentConfig> =
+            AccountLoader::try_from(openbook_v2_fulfillment_config).map_err(|e| {
+                msg!("{:?}", e);
+                ErrorCode::InvalidFulfillmentConfig
+            })?;
+        let openbook_v2_fulfillment_config = load!(openbook_v2_fulfillment_config_loader)?;
+
+        validate!(
+            openbook_v2_fulfillment_config.status == SpotFulfillmentConfigStatus::Enabled,
+            ErrorCode::SpotFulfillmentConfigDisabled
+        )?;
+
+        validate!(
+            &state.signer == drift_signer.key,
+            ErrorCode::InvalidFulfillmentConfig
+        )?;
+
+        validate!(
+            openbook_v2_fulfillment_config.market_index == base_market.market_index,
+            ErrorCode::InvalidFulfillmentConfig,
+            "config market index {} does not equal base asset index {}",
+            openbook_v2_fulfillment_config.market_index,
+            base_market.market_index
+        )?;
+
+        validate!(
+            &base_market.vault == base_market_vault.key,
+            ErrorCode::InvalidFulfillmentConfig
+        )?;
+
+        validate!(
+            &quote_market.vault == quote_market_vault.key,
+            ErrorCode::InvalidFulfillmentConfig
+        )?;
+        validate!(
+            &openbook_v2_fulfillment_config.openbook_v2_program_id == openbook_v2_program.key,
+            ErrorCode::InvalidFulfillmentConfig
+        )?;
+        validate!(
+            openbook_v2_fulfillment_config.market_index == base_market.market_index,
```

Comment:
think you already do the market index check above?

Reply:
duplicate,sorry